arch/arm/core/thread.c

```diff
@@ -67,22 +67,23 @@ void _new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	pInitCtx = (struct __esf *)(STACK_ROUND_DOWN(stackEnd -
 						     sizeof(struct __esf)));
 
+#if CONFIG_ARM_USERSPACE
+	if (!(options & K_USER)) {
```

Comment:
can we swap this so its:

if (options & K_USER) {
   pInitCtx->pc = ((u32_t)_arch_user_mode_enter) & 0xfffffffe;
} else {
   pInitCtx->pc = ((u32_t)_thread_entry) & 0xfffffffe;
}

Reply:
Maybe setting `pInitCtx->pc` to either `_arch_user_mode_enter` or `_thread_entry`, and masking the bits after the `#ifdef` block will generate slightly smaller code.